src/shared/awsagent/policies.go

```diff
@@ -167,31 +198,36 @@ func (a *Agent) SetRolePolicy(ctx context.Context, namespace, accountName string
 	return nil
 }
 
-func (a *Agent) createPolicy(ctx context.Context, role *types.Role, namespace string, intentsServiceName string, statements []StatementEntry) (*types.Policy, error) {
+func (a *Agent) createPolicy(ctx context.Context, role *types.Role, namespace string, intentsServiceName string, statements []StatementEntry, markAsUnusedInsteadOfDelete bool) (*types.Policy, error) {
 	fullPolicyName := a.generatePolicyName(namespace, intentsServiceName)
 	policyDoc, policyHash, err := generatePolicyDocument(statements)
 
 	if err != nil {
 		return nil, errors.Wrap(err)
 	}
 
+	tags := []types.Tag{
+		{
+			Key:   aws.String(policyNameTagKey),
+			Value: aws.String(intentsServiceName),
+		},
+		{
+			Key:   aws.String(policyNamespaceTagKey),
+			Value: aws.String(namespace),
+		},
+		{
+			Key:   aws.String(policyHashTagKey),
+			Value: aws.String(policyHash),
+		},
+	}
+	if markAsUnusedInsteadOfDelete {
```

Comment:
This should be an AWS agent config, set upon initialization (in New()), to avoid problems where one function works like this and the other doesn't.

Reply:
It works this way to support either annotating one pod and setting a global "markAsUnusedInsteadOfDelete" flag